app/models/AppResult.java

```diff
@@ -164,4 +167,19 @@ public static String getSearchFields() {
   public List<AppHeuristicResult> yarnAppHeuristicResults;
 
   public static Finder<String, AppResult> find = new Finder<String, AppResult>(String.class, AppResult.class);
+
+  @Transactional
+  public static int deleteOlderThan(int days, int batchSize, DateTime startingFrom) {
```

Comment:
what do you think of computing a result date in a caller method  ```deleteOlderThan(DateTime jobFinishTime, int batchSize)```?

Reply:
I think it's more self-documenting the way you suggest it.  I was hoping for exactly that.